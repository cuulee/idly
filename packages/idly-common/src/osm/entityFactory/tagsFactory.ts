import { sortObjectKeys } from '../../misc/sortObjectKeys';
import { Tags } from '../structures';

// @TOFIX make tags frozen objects.
// and also figure out how to make reserved keys work in object
// or use so other data structure.
const emptyTag = {};
export function tagsFactory(
  rawTags: { [key: string]: string } = emptyTag
): Tags {
  return sortObjectKeys(rawTags);
}
