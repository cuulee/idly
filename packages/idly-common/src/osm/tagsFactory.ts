import { Tags } from '../osm/structures';

// @TOFIX make tags frozen objects.
// and also figure out how to make reserved keys work in object
// or use so other data structure.
export function tagsFactory(rawTags: Array<[string, string]>): Tags {
  return new Tags(rawTags);
}
