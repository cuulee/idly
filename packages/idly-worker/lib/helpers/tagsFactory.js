export function tagsFactory(obj) {
    const tags = new Map();
    Object.keys(obj).forEach(k => tags.set(k, obj[k]));
    return tags;
}
//# sourceMappingURL=tagsFactory.js.map