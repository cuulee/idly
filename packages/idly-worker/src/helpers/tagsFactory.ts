import { Tags } from 'structs/tags';

export function tagsFactory(obj: { [key: string]: string }): Tags {
  const tags: Tags = new Map();
  Object.keys(obj).forEach(k => tags.set(k, obj[k]));
  return tags;
}
