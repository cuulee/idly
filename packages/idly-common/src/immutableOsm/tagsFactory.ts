import { Tags } from '../osm/structures';
import { $Tags } from '../immutableOsm/immutableOsm';
import { Map as $Map } from 'immutable';

export function $tagsFactory(t: Tags): $Tags {
  return $Map(t);
}
