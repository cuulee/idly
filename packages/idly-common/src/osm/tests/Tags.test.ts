import { Tags } from '../Tags';
import { tagsFactory } from '../tagsFactory';

describe('Tags', () => {
  it('should create instance via tagsFactory', () => {
    const tags = tagsFactory([['1', '2'], ['3', '3']]);
    expect(tags).toBeInstanceOf(Tags);
    const modifiedTags = tags.set('1', 'fourty');
    expect(tags.get('1')).toEqual('2');
    expect(modifiedTags.get('1')).toEqual('fourty');
  });
  it('should stringify', () => {
    const tags = tagsFactory([['1', '2'], ['3', '3']]);
    expect(JSON.stringify(tags)).toMatchSnapshot();
  });
});
