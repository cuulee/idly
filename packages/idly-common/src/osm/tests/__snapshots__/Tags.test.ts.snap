// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should create instance via tagsFactory 1`] = `"\\"[[\\\\\\"1\\\\\\",\\\\\\"2\\\\\\"],[\\\\\\"3\\\\\\",\\\\\\"3\\\\\\"]]\\""`;

exports[`Tags should stringify 1`] = `"\\"[[\\\\\\"1\\\\\\",\\\\\\"2\\\\\\"],[\\\\\\"3\\\\\\",\\\\\\"3\\\\\\"]]\\""`;
